app/controllers/tasks.js

```diff
@@ -33,11 +34,39 @@ export default class TasksController extends Controller {
   @tracked message = ''; // this is required in the modal
   @tracked buttonRequired = false; // this is required in the modal
   @tracked disabled = false; // this is required for the holder component
+  @tracked findingTask = false;
+  @tracked showFetchButton = this.isShowFetchButton();
 
   @action toggleDropDown() {
     this.showDropDown = !this.showDropDown;
   }
 
+  isShowFetchButton() {
+    const inProgressTasks = this.model.filter(
+      (task) => task.status === 'IN_PROGRESS'
+    );
+    const assignedTask = this.model.filter(
+      (task) => task.status === 'ASSIGNED'
+    );
+    return inProgressTasks.length === 0 && assignedTask.length === 0;
+  }
+
+  setShowFetchButton() {
+    this.showFetchButton = this.isShowFetchButton();
+  }
+
+  async setTasksToShow() {
+    try {
+      const response = await fetch(`${API_BASE_URL}/tasks/self`, {
+        credentials: 'include',
+      });
+      const data = await response.json();
+      this.tasksToShow = data;
+    } catch {
```

Comment:
any reason why you discarded the exception value?

Reply:
didn't get you what you are trying to say?

Reply:
not using something like 
`
...
catch(e){
    log(e)
}
`